docs/reference/bpmn-processes/service-tasks/service-tasks.md

```diff
@@ -25,10 +25,11 @@ A service task can define an arbitrary number of `taskHeaders`. They are static
 
 ## Variable mappings
 
-By default, all job variables merge into the process instance. This behavior can be customized by defining an output mapping at the service task.
+Input/output variable mappings can be used to create new variables or customize how variables are merged into the process instance.
 
-Input mappings can be used to transform the variables into a format accepted by the job worker.
+To use the variable mapping, the Zeebe extension element ioMapping must be added to the element. It can contain multiple input and output elements that specify which variables should be mapped. The `Local Input Variable` denotes the variable name inside the activity (a local variable to be created), whereas the `Process Variable Name` of an output denotes the variable name outside of the activity.
```

Comment:
I like where this is going, but:

❌  There is no `Local Input Variable` and no `Process Variable Name`. Please have a look at how the modeler sets the `source` and `target` of the `input` and `output`, when specified. 

```xml
<zeebe:ioMapping>
  <zeebe:input source="= source" target="InputVariable_0in6t79" />
  <zeebe:output source="= source" target="OutputVariable_302pclo" />
</zeebe:ioMapping>
```

Please also update this description in the other pages.

Reply:
In terms of the BPMN XML you are right. But from a users point of view he will see the `Local Input Variable` and `Process Variable Name` in the properties panel. I would not expect him to look in the XML representation. 

In my opinion it would make sense to describe a bit more clearly what variable mapping are from a user point of view. Or we should maybe wipe out these naming differences in the properties panel and BPMN XML.

Reply:
@korthout - any input from your side on my comment here?

Reply:
@Hafflgav I see your point, but not all users use the Camunda Modeler to model their process (e.g. they produce their own XML or use zeebe-bpmn-model to generate it). Zeebe requires XML that adheres to a specific spec, which is currently described in these reference docs. We should not remove that. We can however expand the docs to help out users and I agree that something like that is needed.

Note that the general concept of variable mappings is currently explained in [Concepts](https://docs.camunda.io/docs/components/concepts/variables/#inputoutput-variable-mappings), which also already uses terms like `ioMapping`, `source` and `target`. We should probably change that first. 

AFAIK, the references pages, that this service-task page belongs to, have recently been moved entirely under the Modeler components docs, but IMO not all of it belongs there. My preference would be that the concept is described in the Concept. The Camunda Modeler specific docs on variable mappings should be described in the Modeler component docs, while the way that Zeebe deals with variable mapping should be described in the Zeebe component docs. A guide on using variable mappings should be available in the Guides. I assume that this is also the longer term vision of @akeller and @christinaausley (please correct me if I'm wrong).

As a first step towards that, it might be useful to differentiate in this docs the difference between the concepts that are described (e.g.`Process Variable Name` and `Input Source` or `Input Variable Expression`, etc) and how these are named in different parts (e.g. `source` and `target` in xml and `Local Input Variable` and `Process Variable Name` in the Modeler). Personally I would then first change the concepts page to use the correct terms. We can then change the reference docs on service tasks accordingly. 

How do you see this @Hafflgav, is that the way to go or would you do it differently? Sorry about the long text, I just wanted to share my thoughts to make it clear why I'm against just changing these names only on these references pages, but also provide a way to move forward. I hope it helps.

Reply:
I think at guide for variable mappings make a lot of sense, I'll get that added into our backlog where we can figure out who will write that content. 

Regarding BPMN documentation and it's location, I'm happy to talk more about this separately.